src/pages/ThemeProductListPage.tsx

```diff
@@ -0,0 +1,154 @@
+  color: ${colors.white};
+`;
+
+const ProductListContainer = styled.div`
+  display: grid;
+  grid-template-columns: repeat(2, 1fr);
+  gap: ${spacing.lg};
+  padding: ${spacing.lg};
+`;
+
+const LoadMoreTrigger = styled.div`
+  height: 20px;
+  width: 100%;
+  grid-column: 1 / -1;
+`;
+
+const EmptyState = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  padding: ${spacing.xl};
+  color: ${colors.gray600};
+  text-align: center;
+  grid-column: 1 / -1;
+`;
+
+export const ThemeProductListPage = () => {
+  const { themeId } = useParams();
+  const navigate = useNavigate();
+  const { themeInfo, loading, error, is404Error } = useFetchThemeInfo(themeId || '');
+  const {
+    themeProduct,
+    loading: themeProductLoading,
+    error: themeProductError,
+    hasMoreList,
+    loadMore,
+  } = useFetchThemeProduct(themeId || '');
+
+  const observerRef = useRef<IntersectionObserver | null>(null);
+  const loadMoreRef = useRef<HTMLDivElement>(null);
+
+  const handleObserver = useCallback(
+    (entries: IntersectionObserverEntry[]) => {
+      const [target] = entries;
+      if (target.isIntersecting && hasMoreList && !themeProductLoading) {
+        loadMore();
+      }
+    },
+    [hasMoreList, themeProductLoading, loadMore],
+  );
+
+  useEffect(() => {
+    const element = loadMoreRef.current;
+    if (!element) return;
+
+    observerRef.current = new IntersectionObserver(handleObserver, {
+      threshold: 0.1,
+    });
+
+    observerRef.current.observe(element);
+
+    return () => {
+      if (observerRef.current) {
+        observerRef.current.disconnect();
+      }
+    };
+  }, [handleObserver]);
+
+  // 404 에러 처리
+  useEffect(() => {
+    if (is404Error) {
+      navigate('/');
+    }
+  }, [is404Error, navigate]);
+
+  return (
+    <AppContainer>
+      <Header title="선물하기" />
+      {loading ? (
+        <Loading />
+      ) : error ? (
+        <div>테마 정보 불러오기 실패</div>
+      ) : (
+        <ThemeInfoContainer backgroundColor={themeInfo?.backgroundColor || ''}>
+          <ThemeInfoName>{themeInfo?.name}</ThemeInfoName>
+          <ThemeInfoTitle>{themeInfo?.title}</ThemeInfoTitle>
+          <ThemeInfoDescription>{themeInfo?.description}</ThemeInfoDescription>
+        </ThemeInfoContainer>
+      )}
+      <ProductListContainer>
+        {themeProductLoading && themeProduct.length === 0 ? (
+          <Loading />
+        ) : themeProductError ? (
+          <div>테마 제품 불러오기 실패</div>
+        ) : themeProduct.length === 0 ? (
+          <EmptyState>
+            <div>상품이 없습니다</div>
+          </EmptyState>
+        ) : (
+          <>
+            {themeProduct.map((product) => (
+              <ProductCard key={product.id} product={product} />
+            ))}
+            <LoadMoreTrigger ref={loadMoreRef}>
+              {themeProductLoading && hasMoreList && <Loading />}
+            </LoadMoreTrigger>
+          </>
+        )}
```

Comment:
사람마다 다르긴 한데, 저는 보통 요런 복잡한 삼항연산자 보다는 즉시실행함수 + early return을 선호하는데요!...
조건을 더 쉽게 파악하고, 어떤 케이스에 어떤 데이터(jsx)가 리턴되는지 더 쉽게 파악할 수 있답니다!

요건 참고만 해주세요 ㅎㅎ
 ```suggestion
        {(() => {
          if (themeProductLoading && themeProduct.length === 0) {
            return <Loading />;
          }
          if (themeProductError) {
            return <div>테마 제품 불러오기 실패</div>;
          }
          if (themeProduct.length === 0) {
            return (
              <EmptyState>
                <div>상품이 없습니다</div>
              </EmptyState>
          );
          }
          return <></>;
        })()}
```